server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/MirrorSchedulingService.java

```diff
@@ -209,9 +221,21 @@ private void schedulePendingMirrors() {
                               if (m.schedule() == null) {
                                   continue;
                               }
+                              if (zoneConfig != null) {
+                                  String pinnedZone = m.zone();
+                                  if (pinnedZone == null) {
+                                      // Use the first zone if the mirror does not specify a zone.
+                                      pinnedZone = zoneConfig.allZones().get(0);
+                                  }
+                                  if (!pinnedZone.equals(currentZone)) {
+                                      // Skip the mirror if the zone does not match.
```

Comment:
Question) Would it be worth checking if the `pinnedZone` is contained in `ZoneConfig#allZones` to detect ill-configured mirrors?

Reply:
Most users may choose the `pinnedZone` from the UI's select box, but it also makes sense to validate on the server side.

Reply:
Sorry, let me rephrase my concern.

For instance, if servers are deployed in JP1, JP2, JP3 and we decide to migrate (JP3 -> JP4), is there no need for us to be aware of failing mirror jobs? I think moving servers across zones is not such a rare occasion.

Reply:
It makes more sense. Let me report a failure to `MirrorListener`.

Reply:
Note that I also want to generalize the mirror listener so that it also reports the when a k8s aggregator fails.

Reply:
Are you planning to design a general listener to receive arbitrary events?

Reply:
Yes. 😄

Reply:
It sounds nice. I'm looking forward to the Central Dogma event bus. 🚌